src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1+Builder.cs

```diff
@@ -636,6 +636,25 @@ public void Sort()
             }
 
             /// <summary>
+            /// Sorts the elements in the entire array using
+            /// the specified <see cref="Comparison{T}"/>.
+            /// </summary>
+            /// <param name="comparison">
+            /// The <see cref="Comparison{T}"/> to use when comparing elements.
+            /// </param>
+            /// <exception cref="ArgumentNullException"><paramref name="comparison"/> is null.</exception>
+            [Pure]
+            public void Sort(Comparison<T> comparison)
+            {
+                Requires.NotNull(comparison, "comparison");
+
+                if (Count > 1)
+                {
+                    Array.Sort(_elements, comparison);
```

Comment:
@AArnott, this is accessing both Count and _elements... do we need to do the whole ```var self = this;``` thing?